hoot-core/src/main/cpp/hoot/core/io/OsmApiNetworkRequest.cpp

```diff
@@ -93,8 +93,13 @@ bool OsmApiNetworkRequest::networkRequest(QUrl url, QNetworkAccessManager::Opera
   loop.exec();
   //  Get the status and content of the reply if available
   _status = _getHttpResponseCode(reply);
-  if (reply != NULL)
-    _content = reply->readAll();
+  //  According to the documention this shouldn't ever happen
+  if (reply == NULL)
+  {
+    throw HootException(QString("Network request error: GET/POST/PUT request failed to create reply object."));
+    return false;
+  }
```

Comment:
A `NULL` reply is a very bad thing.  It should only happen when there is no memory to allocate the object.